File: solutions/javascript/reverse-string/1/reverse-string.js

export const reverseString = (str) => {
  const chars = str.split('');
  const middle = Math.floor(chars.length / 2);
  for (let i = 0; i < middle; i++) {
    const temp = chars[i];
    chars[i] = chars[chars.length - 1 - i];
    chars[chars.length - 1 - i] = temp;
  }
  return chars.join("");
};

File: solutions/cpp/reverse-string/1/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string
{
  std::string reverse_string(std::string str)
  {
    int middle = str.length() / 2; 
    for (int i = 0; i < middle; i++)
    {
      char temp = str[i];
      str[i] = str[str.length() - 1 - i];
      str[str.length() - 1 - i] = temp;
    }
    return str;
  }
}  // namespace reverse_string
